docs/testnet-guardian-runbook.md

```diff
@@ -0,0 +1,152 @@
+The nodes are configured to restart upon crash for some of the known issues (#4295). Create issues for new ones.
+
+2. Run `scripts/ezcheck.sh` locally by passing the namespace to check if all the nodes are synced and have the same state hash at their best tips. This requires kubectl and gcloud authentication.
+
+3. Query the logs to confirm that the nodes are creating and/or receiving blocks and that the blocks include proofs and transactions.
+Some strings to look for: `Received a block`, `apply_diff block info`, `Number of proofs ready for purchase:`, `Detailed diff creation log:`
+
+4. Connect an external node to the network and check if it syncs.
+
+Publish the multiaddresses of the seed nodes on `#QA-net details` channel for the rest of the team to connect to the QA-net.
+Seeds nodes can be looked up here: https://console.cloud.google.com/compute. Usually there are two seed nodes per tesnet, click on the seed node instance to go to the page that has details about the instance required to ssh into it.
+<img src="res/gcloud_ssh.png" alt="ssh info" width="300"/>
+
+`docker exec -it <container> /bin/bash` into coda-daemon container and follow [this](#multiaddress) to get the multiaddress of the node. These are then published on the website (https://codaprotocol.com/docs/getting-started) for users to connect to the testnet.
+
+Alternatively, multiaddresses of seed nodes can be obtained from the daemon command of other nodes in the testnet which use them as initial peers (using `-peer` flag). Retrieve the daemon command using: 
+`kubectl exec -it <pod-id> --namespace <namespace> -c <container-name> cat /proc/10/cmdline`
+
+ Any other node in the testnet can also be used as an initial peer to connect to the testnet. To obtain the multiaddress of non-seed nodes that are running on kubernetes, run `kubectl exec -it <pod-id> /bin/bash --namespace <namespace> -c <container-name>` and follow [this](#multiaddress).
+
+##### Multiaddress
+
+Run `coda client status` to get the IP address [IPADDR], Libp2p port [PORT] and peer id [PEERID]  and combine them in the following format:
+    `/ip4/IPADDR/tcp/PORT/ipfs/PEERID`
+For example: `/ip4/104.196.41.236/tcp/10001/ipfs/12D3KooWAFFq2yEQFFzhU5dt64AWqawRuomG9hL8rSmm5vxhAsgr`
+
+##### Complete the challenges for the release
+
+Connect to the QA-net from your machine and complete the protocol/product related challenges for the release. This is to test the features required for the challenges are working and also, may help the testnet guardian prepare for the kind of questions the community may have after the release.
+Note: Any challenge that is resource intensive like "make most snark works" should ideally be part of the QA-net configuration. For example: run multiple snark workers with different keys and fees to verify if they have been included. Alternatively, the guardian could run a snark worker locally for a couple of hours and check how many of them were included.
+
+##### Check docs
```

Comment:
Some of these are very specifically for the product team, like this one, and some for protocol.  Does it make sense to specify which tasks are for which team?  Possibly tag them or split them into sections?

Reply:
I did not distinguish between product/protocol tasks as such. The guardian basically checks if the docs are up-to-date with the release. Were you thinking the guardian would also update the docs (in which case i'll have to add instructions for those)? Also, would we have a product and a protocol engineer per release? I was imagining it to be two members from either from the product or the protocol team. Configuring and validating the QA net would need some collaboration in the beginning but otherwise, I think the tasks can be performed by anyone on the engineering team?

Reply:
Talking about it some more, this approach makes sense.  Will update the primary and the secondary, can be from either team.  That will help close the knowledge gap on both sides.  Great point!